### PR TITLE
Clarify cps/constants.py and scripts/lb-wrapper [& refine URL validation, to accept http:// legacy URLs w/o SSL/TLS]

### PR DESCRIPTION
**Pull request by holta**
@deldesir: ok to tighten up a bit for readability?

Can you remind if/why `if [[ ! ${URL} =~ ^http[s]?:// ]]` is truly necessary in lb-wrapper?  (Do xklb and/or yt-dlp fail if $URL doesn't begin with "http://" or "https://" ?)

### CODE REVIEW

File: scripts/lb-wrapper [hunk 1]
```diff
@@ -2,44 +2,41 @@
...
 
 # Function to log messages
 log() {
     echo "$(date +'%Y-%m-%d %H:%M:%S') - $1" | tee -a ${LOG_FILE}
 }
 
 # Check if xklb is installed
-if ! command -v "${XKLB_EXECUTABLE}" &> /dev/null
-then
+if ! command -v "${XKLB_EXECUTABLE}" &> /dev/null; then
```
Comment (holta) [inline scripts/lb-wrapper:18]: @ddeldesir I don't think `&> /dev/null;` does much anything in this case?

```suggestion
if ! command -v "${XKLB_EXECUTABLE}"; then
```

(Then again it's also harmless...)

**Comment by holta** [comment]
> Can you remind if/why `if [[ ! ${URL} =~ ^http[s]?:// ]]` is truly necessary in lb-wrapper? (Do xklb and/or yt-dlp fail if $URL doesn't begin with "http://" or "https://" ?)

@deldesir the 2 URL validation sections look stale, can you confirm/suggest alongside f50f102 ?

https://github.com/iiab/calibre-web/blob/6ed862a65df74030890fe62d8e90b07d3ab7faaa/cps/static/js/main.js#L182-L188

https://github.com/iiab/calibre-web/blob/6ed862a65df74030890fe62d8e90b07d3ab7faaa/cps/static/js/main.js#L275-L281

**Comment by holta** [comment]
> @deldesir these 2 URL validation sections look stale, can you confirm/suggest alongside https://github.com/iiab/calibre-web/commit/f50f102b0a055e239bd31ff5c2c591e3eb46a565 ?

Possibly too difficult today, but let's think about unifying URL validation into 1 single place if possible?

(Rather than 3 different places, making it much harder to sanity-check and maintain!)

File: scripts/lb-wrapper (hunk 1 above)
Comment (deldesir) [reply scripts/lb-wrapper:18]: Ok, I get it.

**Comment by deldesir** [comment]
> @deldesir: ok to tighten up a bit for readability?
> 
> Can you remind if/why `if [[ ! ${URL} =~ ^http[s]?:// ]]` is truly necessary in lb-wrapper?  (Do xklb and/or yt-dlp fail if $URL doesn't begin with "http://" or "https://" ?)

I remember with a previous version of xklb, the script failed due to the missing protocol. I have yet to test with latest version.

**Comment by holta** [comment]
> let's think about unifying URL validation into 1 single place if possible?
>
> (Rather than 3 different places, making it much harder to sanity-check and maintain!)

FWIW this PR narrows it from 3 different places, to just 2 (possibly that's good enough for now?)

Finally, this PR was tested successfully on a very recent Ubuntu 24.04 pre-release.

File: cps/static/js/main.js [hunk 2]
```diff
@@ -179,13 +179,19 @@ $(document).ready(function() {
...
         url = url.startsWith("https://") ? url : "https://" + url;
         if (!isValidURL(url)) {
             alert("Invalid URL");
             return;
         }
+        */
+
+        // Prepend https:// if URL doesn't begin with http:// or https://
+        // As xklb requires: https://github.com/iiab/calibre-web/pull/44
+        url = /^https?:\/\//.test(url) ? url : "https://" + url;
```
Comment (deldesir) [inline cps/static/js/main.js:194]: Personally, I prefer string manipulation over regex.

**Review by deldesir — APPROVED** [review]
Looks good to me.

File: cps/static/js/main.js (hunk 2 above)
Comment (holta) [reply cps/static/js/main.js:194]: To be clear:

This PR fixes support for legacy (e.g. bookmarked) URLs that begin with **http://** (i.e. without SSL / TLS).

**Comment by holta** [comment]
Subject line changed, to clarify that this PR fixes support for legacy (e.g. bookmarked) URLs that begin with http:// (i.e. without SSL / TLS).

**Comment by holta** [comment]
> Subject line changed, to clarify that this PR fixes support for legacy (e.g. bookmarked) URLs that begin with http:// (i.e. without SSL / TLS).

CLARIF:

Yes it could be argued that we really should also tolerate URLs that begin with uppercase letters in the protocol...

- HTTP
- HtTp
- ...
- HTTPS
- hTtPs
- ...
- ETC

(But the above corner cases are extremely rare, and we have Far Bigger Fish to fry!)

**Pull request merged**